bundles/model/org.eclipse.smarthome.model.sitemap/src/org/eclipse/smarthome/model/sitemap/internal/SitemapProviderImpl.java

```diff
@@ -28,55 +37,77 @@
  * @author Kai Kreuzer - Initial contribution and API
  *
  */
-public class SitemapProviderImpl implements SitemapProvider {
+@Component(service = SitemapProvider.class)
+public class SitemapProviderImpl implements SitemapProvider, ModelRepositoryChangeListener {
 
-    protected static final String SITEMAP_FILEEXT = ".sitemap";
+    private static final String SITEMAP_MODEL_NAME = "sitemap";
+    protected static final String SITEMAP_FILEEXT = "." + SITEMAP_MODEL_NAME;
 
     private final Logger logger = LoggerFactory.getLogger(SitemapProviderImpl.class);
 
     private ModelRepository modelRepo = null;
 
+    private final Map<String, Sitemap> sitemapModelCache = new ConcurrentHashMap<>();
+
+    @Reference(cardinality = ReferenceCardinality.MANDATORY, policy = ReferencePolicy.STATIC)
     public void setModelRepository(ModelRepository modelRepo) {
         this.modelRepo = modelRepo;
+        modelRepo.addModelRepositoryChangeListener(this);

```

Comment:
This can potentially lead to callbacks even before this component is activated.
Even though we do not care today about the activation state in this component, I would prefer to add the listener in the activate method (to be created) only.